### DISCUSSION TITLE
New packages to handle queue and Jobs in the core?

### DISCUSSION THREAD

**Discussion by Saeeed-B** [meteor/meteor#11824] [Feature Requests]
I think there should be a queue and job system in the meteor core.
This is a powerful tool for the meteor.

**Comment by Saeeed-B** [comment]
I want to be clear.
Problems with NodeJS frameworks , start as their dependence on external packages increases.
You need to launch the essentials of a large and complete software in your Meteor kernel.

**Comment by Saeeed-B** [comment]
Look at Laravel.
The Queues and Jobs tool is a very basic tool.
Why not in Meteor?
With none of these vital tools in Meteor, more users are giving up on meteor.

**Reply by StorytellerCZ** [reply]
While I agree with your general sentiment, a bit more is needed. There are multiple packages for jobs, each working a bit differently and suiting a different need. So first what we have to overcome is why this should be in core and take more resources from developing other parts of Meteor? After that which methodology and approach should be used for this?

**Reply by Saeeed-B** [reply]
meteor should be able to queue tasks such as sending email and sending SMS.

In Laravel, almost anything can be queued. Even creating a database record.
Like creating a notification record for users in the database.

**Comment by Saeeed-B** [comment]
Many NodeJS frameworks have the problem that external packages that play important roles have a direct impact on that framework.
For example, in react native, there were packages that disrupted many programs when their support ran out
  And users have migrated to Flutter.

I want this to be understood in Meteor.

**Comment by Saeeed-B** [comment]
Basically, in Meteor, the policy should be that the developers focus on development and debugging easily and quickly.
But since entering Meteor, I have focused only on solving basic problems instead of developing my software.

**Comment by filipenevola** [comment]
Hi @Saeeed-B we should avoid spamming the same content everywhere.

Opening a discussion is the right way to ask for a feature but sending a ton of messages is not. One message should be enough until you have a reply. GitHub supports editing messages. Also duplicating it in two different topics in the Forums is not good.

As @StorytellerCZ said, why do you think it should be in the core?

We have many supported and working packages for this exact same purpose.

For example, I use https://github.com/percolatestudio/meteor-synced-cron in all my projects without any issues.

What is missing in the existent packages that should come from the core?

Please be sure you review this [text](https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests) with attention as well. There we explain the guideline for feature requests.

**Comment by filipenevola** [comment]
> I think there should be a queue and job system in the meteor core. This is a powerful tool for the meteor.

Your title was saying one thing and your description a different thing, so I changed your title.

**Comment by Saeeed-B** [comment]
I mean exactly the queuing in the database.
For example, we want to send emails to 100 of our users, and these 100 operations are transferred to the database and executed in order.
I do not want to use Corn Job
